handlers/room_memberships_handler.go

```diff
@@ -24,12 +35,7 @@ func RoomMembershipsCreate(c *gin.Context) {
 		UserId: user.Id,
 	}
 
-	membership, err := models.FindOrCreateRoomMembership(r)
-	if err != nil {
-		c.Fail(500, err)
-	}
-
-	c.JSON(200, gin.H{"membership": membership})
+	return models.FindOrCreateRoomMembership(r)
```

Comment:
So the one thing here is that this will automatically subscribe the user to notifications for this room, and then below the call to `DeleteRoomMembership` will prevent them from receiving notifications when they aren't in the room.

I probably should have named the RoomMembership model and its related methods better, but it's mostly for receiving updates when the user is away from the room; I think we might want something more ephemeral for handling the socket connections specifically.